src/main/java/mate/academy/service/UserService.java

```diff
@@ -1,20 +1,24 @@
 package mate.academy.service;
 
+import java.util.Arrays;
 import mate.academy.model.User;
 
 public class UserService {
-    private static final User[] users = new User[] {
+    private static final User[] users = new User[]{
             new User("bob@i.ua", "1234"),
             new User("alice@i.ua", "1234")
     };
 
     /**
      * Find user by email. All users are stored in <code>private static final User[] users</code>
+     *
      * @param email - the input parameter
      * @return - user if his email is equal to passed email.
      * Return <code>null</code> if there is no suitable user
      */
-    public User findByEmail(String email) {
-        return null;
+    public static User findByEmail(String email) {
+        User user = Arrays.stream(users).filter(findUser -> findUser.getEmail().equals(email))
+                .findFirst().orElse(null);
+        return user != null ? user : new User("none", "none");
```

Comment:
It's kind of hardcoding, if user doesn't exist you should return `null` and not create a new one. If you already figure out what is `Optional`, you can return Optional.of(user), which your stream actually returns, or you can just return user, so it will be `null` if there is no user with such an email. Also, don't forget to check if user is `null` in `login` method